### PR TITLE
rename `coder` to `code_server`

### PR DESCRIPTION
**Pull request by GiorgioBrux**
# Description

Renamed `coder` to `code_server` since `coder` is now a different software.
The old `coder` folder would not be automatically renamed, so anyone using the old role would have to manually rename it and then run the new tag.

# How Has This Been Tested?

Even if it's a simple rename I tested it and it works as expected.

### CODE REVIEW

**Comment by saltydk** [comment]
Since this is a breaking change it should migrate the old opt folder or have some kind of method to deal with that.

**Comment by saltydk** [comment]
Maybe @owine can help you with how to make the migration specific to this role and not moving the opt folder if you use the 'real' coder.

**Comment by GiorgioBrux** [comment]
Yep, I wasn't totally sure how to handle the migration assuming someone could be already running the real coder and using the coder folder. I'll try looking into it later.

**Comment by GiorgioBrux** [comment]
Let me know if I should change/add anything else.

**Comment by owine** [comment]
```
[yaml[indentation]: roles/code_server/tasks/subtasks/legacy.yml#L36](https://github.com/saltyorg/Sandbox/pull/207/files#annotation_7276666352)
Wrong indentation: expected 8 but found 10
```

**Comment by GiorgioBrux** [comment]
> ```
> [yaml[indentation]: roles/code_server/tasks/subtasks/legacy.yml#L36](https://github.com/saltyorg/Sandbox/pull/207/files#annotation_7276666352)
> Wrong indentation: expected 8 but found 10
> ```

Oops, fixed.

File: roles/coder/tasks/main.yml [hunk 1]
```diff
@@ -1,24 +1,12 @@
...
-  ansible.builtin.include_tasks: "{{ resources_tasks_path }}/docker/remove_docker_container.yml"
-
-- name: Create directories
-  ansible.builtin.include_tasks: "{{ resources_tasks_path }}/directories/create_directories.yml"
-
-- name: Create Docker container
-  ansible.builtin.include_tasks: "{{ resources_tasks_path }}/docker/create_docker_container.yml"
+- name: "Fail with deprecation warning"
+  ansible.builtin.fail:
+    msg: "The 'coder' role which used the 'code-server' image is currently deprecated in favour of the 'code-server' role using the same image. If you are using the default '/opt/coder' folder, after renaming the custom variables in the inventory if you have any, run \"sb install sandbox-code_server -e 'code_server_migrate_coder=true'\" once to move the files over to the new folder', then simply use the 'code-server' role from now on."
```
Comment (owine) [inline roles/coder/tasks/main.yml:12]: We need `when: not continuous_integration` on this task so CI goes green

**Comment by GiorgioBrux** [comment]
Making it a draft again so I can fix the CI not passing.

File: roles/code_server/tasks/subtasks/legacy.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    - name: Legacy | Set permissions on 'code_server'
+      ansible.builtin.file:
+        path: "{{ code_server_paths_location }}"
+        state: directory
+        recurse: yes
+        owner: "{{ user.name }}"
+        group: "{{ user.name }}"
+
+  when: (code_server_migrate_coder)
```
Comment (owine) [inline roles/code_server/tasks/subtasks/legacy.yml:56]: Remove conditional here

File: roles/code_server/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+# Title:            Sandbox: code-server                                #
+# Author(s):        astroMD, kowalski                                   #
+# URL:              https://github.com/saltyorg/Sandbox                 #
+# --                                                                    #
+#########################################################################
+#                   GNU General Public License v3.0                     #
+#########################################################################
+---
+- name: Import Legacy task
+  ansible.builtin.import_tasks: "subtasks/legacy.yml"
```
Comment (owine) [inline roles/code_server/tasks/main.yml:11]: Move the conditional check to here and use `when: code_server_migrate_coder is defined` so we can use the var without having to define it beforehand

**Review by owine — CHANGES_REQUESTED** [review]

**Comment by owine** [comment]
```
[yaml[trailing-spaces]: roles/code_server/tasks/subtasks/legacy.yml#L29](https://github.com/saltyorg/Sandbox/pull/207/files#annotation_7279686803)
Trailing spaces

[yaml[empty-lines]: roles/code_server/tasks/subtasks/legacy.yml#L54](https://github.com/saltyorg/Sandbox/pull/207/files#annotation_7279686814)
Too many blank lines (1 > 0)

[yaml[new-line-at-end-of-file]: roles/coder/tasks/main.yml#L13](https://github.com/saltyorg/Sandbox/pull/207/files#annotation_7279686820)
No new line character at the end of file
```

**Comment by GiorgioBrux** [comment]
My god I fix something and break something else, apologies, 1 minute.

File: roles/coder/tasks/main.yml [hunk 4]
```diff
@@ -1,24 +1,13 @@
...
+    when: not continuous_integration
```
Comment (owine) [inline roles/coder/tasks/main.yml:13]: This is on the wrong indent - it should be on the same indent as `name` and `ansible.builtin.fail`

**Review by owine — CHANGES_REQUESTED** [review]

File: roles/coder/tasks/main.yml (hunk 4 above)
Comment (owine) [reply roles/coder/tasks/main.yml:13]: This should be the last thing

**Review by owine — APPROVED** [review]

**Pull request merged**